src/common/lockdep.cc

```diff
@@ -261,6 +261,10 @@ int lockdep_will_lock(const char *name, int id)
 
 	assert(0);  // actually, we should just die here.
       } else {
+        BackTrace *bt = NULL;
+        if (g_lockdep >= 2 || force_backtrace) {
```

Comment:
Just following the existing logic with ```g_lockdep >= 2``` -- I assumed somebody was forcing it to 2 explicitly outside of the config.  I see in ```common_init.cc``` it is explicitly set to true if the config lockdep option is enabled.  Perhaps it would be cleaner with a new "lockdep_force_backtrace" config value?